### PR TITLE
Add safety to Vector, RotMatrix math functions

### PR DESCRIPTION
**Pull request by noche-x**

### CODE REVIEW

File: RosaServer/api.cpp [hunk 1]
```diff
@@ -1122,42 +1123,50 @@ std::string Vector::__tostring() const {
...
 	return {x + other->x, y + other->y, z + other->z};
 }
 
 Vector Vector::__sub(Vector* other) const {
+	if (!other) throw std::invalid_argument(missingArgument);
 	return {x - other->x, y - other->y, z - other->z};
 }
 
 Vector Vector::__mul(float scalar) const {
+	if (!scalar) throw std::invalid_argument(missingArgument);
```
Comment (jdbool) [inline RosaServer/api.cpp:1136]: Not a pointer, shouldn't check this

File: RosaServer/api.cpp [hunk 2]
```diff
@@ -1122,42 +1123,50 @@ std::string Vector::__tostring() const {
...
 
 Vector Vector::__mul_RotMatrix(RotMatrix* rot) const {
+	if (!rot) throw std::invalid_argument(missingArgument);
 	return {rot->x1 * x + rot->y1 * y + rot->z1 * z,
 	        rot->x2 * x + rot->y2 * y + rot->z2 * z,
 	        rot->x3 * x + rot->y3 * y + rot->z3 * z};
 }
 
 Vector Vector::__div(float scalar) const {
+	if (!scalar) throw std::invalid_argument(missingArgument);
```
Comment (jdbool) [inline RosaServer/api.cpp:1148]: Not a pointer

File: RosaServer/api.cpp [hunk 3]
```diff
@@ -1122,42 +1123,50 @@ std::string Vector::__tostring() const {
...
 
 void Vector::add(Vector* other) {
+	if (!other) throw std::invalid_argument(missingArgument);
 	x += other->x;
 	y += other->y;
 	z += other->z;
 }
 
 void Vector::mult(float scalar) {
+	if (!scalar) throw std::invalid_argument(missingArgument);
```
Comment (jdbool) [inline RosaServer/api.cpp:1162]: Not a pointer

**Review by jdbool — CHANGES_REQUESTED** [review]

File: RosaServer/api.cpp (hunk 1 above)
Comment (noche-x) [reply RosaServer/api.cpp:1136]: does sol make the base type params not null if not given

Comment (jdbool) [reply RosaServer/api.cpp:1136]: There's no such thing as a null float

**Comment by noche-x** [comment]
I removed the lines that didn't make sense. need anymore changes?

**Comment by noche-x** [comment]
I forgot to push my changes...

File: RosaServer/rosaserver.h [hunk 4]
```diff
@@ -4,6 +4,7 @@
 #include <iostream>
 #include <thread>
 
+#define SOL_SAFE_FUNCTION_CALLS 1
```
Comment (jdbool) [inline RosaServer/rosaserver.h:7]: Don't define this, too much overhead

**Review by jdbool — CHANGES_REQUESTED** [review]

**Pull request merged**